File: 0137-single-number-ii/solution.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        hm = defaultdict(int)

        for i in nums:
            hm[i]+=1
            
            if hm[i] == 3:
                hm.pop(i)

        return list(hm.keys())[0]
        

File: 0137-single-number-ii/solution.java
class Solution {
    public int singleNumber(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        
        for (int x : nums) {
            map.put(x, map.getOrDefault(x, 0) + 1);
        }

        for (int num: map.keySet()) {
            if (map.get(num) == 1) {
                return num;
            }
        }
        return -1;
    }
}
